src/openforms/appointments/contrib/jcc/plugin.py

```diff
@@ -71,12 +80,47 @@ def get_available_products(
             for entry in result
         ]
 
+    def _get_all_locations(self, client: Client) -> list[AppointmentLocation]:
+        try:
+            location_ids = client.service.getGovLocations()
+        except (ZeepError, RequestException) as e:
+            logger.exception("Could not retrieve location IDs", exc_info=e)
+            return []
+        except Exception as exc:
+            raise AppointmentException from exc
+
+        with parallel() as pool:
+            details = pool.map(
+                lambda location_id: client.service.getGovLocationDetails(
+                    locationID=location_id
+                ),
+                location_ids,
+            )
+
+        locations = [
+            AppointmentLocation(
+                identifier=identifier,
+                name=entry["locationDesc"],
+                address=entry["address"],
+                postalcode=entry["postalcode"],
+                city=entry["city"],
+            )
+            for identifier, entry in zip(location_ids, details)
+        ]
```

Comment:
Should this also be wrapped in a try/except like the `getGovLocations` calls so in case of errors in also raises the appropriate errors?

Reply:
ah frick, it should... this exception handling is very un-elegant :(

I'll add a test for it tomorrow and exception handling.